lib/api.go

```diff
@@ -0,0 +1,127 @@
+package sebak
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"boscoin.io/sebak/lib/error"
+	"boscoin.io/sebak/lib/storage"
+	"github.com/GianlucaGuarini/go-observable"
+	"github.com/gorilla/mux"
+)
+
+type HandlerFunc func(w http.ResponseWriter, r *http.Request)
+
+func AddHandlers(r interface{}, ctx context.Context) {
+	if r == nil {
+		return
+	}
+	router, ok := r.(*mux.Router)
+	if !ok {
+		return
+	}
+	router.HandleFunc("/account/{address}", GetAccountHandler(ctx)).Methods("GET")
+}
+
+func streaming(o *observable.Observable, w http.ResponseWriter, event string, callBackFunc func(args ...interface{}) ([]byte, error), once []byte) {
+	cn, ok := w.(http.CloseNotifier)
+	if !ok {
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	closeChan := make(chan bool)
+	messageChan := make(chan []byte)
+
+	observerFunc := func(args ...interface{}) {
+		s, err := callBackFunc(args...)
+		if err != nil {
+			closeChan <- true
+			return
+		}
+		messageChan <- s
+	}
+
+	o.On(event, observerFunc)
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if len(once) != 0 {
+		fmt.Fprintf(w, "%s\n", once)
+		flusher.Flush()
+	}
+
+	for {
+		select {
+		case <-cn.CloseNotify():
+			o.Off(event, observerFunc)
+			return
+		case <-closeChan:
+			o.Off(event, observerFunc)
+			return
+		case message := <-messageChan:
+			fmt.Fprintf(w, "%s\n", message)
+			flusher.Flush()
+		}
+	}
+}
+
+func GetAccountHandler(ctx context.Context) HandlerFunc {
+	storage, ok := ctx.Value("storage").(*sebakstorage.LevelDBBackend)
```

Comment:
IMHO,  Using `LevelDBBackend` with `context.Context.Value` is not suitable in normal cases.

Reply:
Issue what you picked is spread all around of handlers for sebak.
I think we need to talk about this in another issue.